aspnetcore/fundamentals/middleware/extensibility/sample/Startup.cs

```diff
@@ -0,0 +1,41 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using MiddlewareExtensibilitySample.Data;
+using MiddlewareExtensibilitySample.Middleware;
+
+namespace MiddlewareExtensibilitySample
+{
+    public class Startup
+    {
+        #region snippet1
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddDbContext<AppDbContext>(options =>
+                options.UseInMemoryDatabase("InMemoryDB"));
+            services.AddScoped<IMiddlewareFactory, BasicMiddlewareFactory>();
```

Comment:
I would have 2 examples:
1. Showing the default IMiddlewareFactory
1. Showing how you replace the IMiddlewareFactory for something like simple injector

Reply:
It doesn't sound like "default IMiddlewareFactory" refers to the conventional approach. By "default MiddlewareFactory," do you mean what I show here as the "BasicMiddlewareFactory"?

Reply:
Nevermind ... I see what you mean now. The `IMiddlewareFactory` itself (not replaced) is the default, and you want to show the default SxS with the `BasicMiddlewareFactory` here.